flink-end-to-end-tests/flink-connector-hive-test/src/test/java/org/apache/flink/connectors/hive/HiveConnectorITCase.java

```diff
@@ -0,0 +1,171 @@
+						testJarPath);
+				jobSubmissionBuilder.setParallelism(1)
+						.addOption("-ys", "1")
+						.addOption("-ytm", "1000")
+						.addOption("-yjm", "1000")
+						.addOption("-c", HiveReadWriteDataExample.class.getCanonicalName())
+						.addArgument("--hiveVersion", hiveVersion)
+						.addArgument("--sourceTable", sourceTable)
+						.addArgument("--targetTable", destTable);
+				YarnClusterFlinkResource.YarnClusterJobController jobController =
+						(YarnClusterFlinkResource.YarnClusterJobController) clusterController.submitJob(
+								jobSubmissionBuilder.build());
+				log.info(jobController.fetchExecuteLog());
+			}
+			String actualResults = clusterAndHiveResource.execHiveSql(String.format("select * from %s", destTable));
+			Assert.assertEquals(expectedResults, actualResults);
+		}
+	}
+
+	/**
+	 * Test all dataTypes now we supported.
+	 * @throws Exception
+	 */
+	@Test
+	public void testComplexReadWriteHiveTable() throws Exception {
+		clusterAndHiveResource.execHiveSql(
+				"CREATE TABLE `all_types_table`(\n" +
+				"  `tinyintcol` tinyint,\n" +
+				"  `samllinttypecol` smallint,\n" +
+				"  `intcol` int,\n" +
+				"  `bigintcol` bigint,\n" +
+				"  `floatcol` float,\n" +
+				"  `doublecol` double,\n" +
+				"  `decimalcol` decimal(10,0),\n" +
+				"  `decimalprecisioncol` decimal(38,10),\n" +
+				"  `timestampcol` timestamp,\n" +
+				"  `datecol` date,\n" +
+				"  `varcharcol` varchar(20),\n" +
+				"  `charcol` char(10),\n" +
+				"  `stringcol` string,\n" +
+				"  `booleancol` boolean,\n" +
+				"  `binarycol` binary,\n" +
+				"  `arraycol` array<struct<a:int,t:string>>,\n" +
+				"  `mapcol` map<int,array<string>>,\n" +
+				"  `structcol` struct<a:tinyint,b:char(10),m:map<int,string>>)\n" +
+				"COMMENT 'all datatypes table'\n" +
+				"ROW FORMAT DELIMITED FIELDS TERMINATED BY ';' COLLECTION ITEMS TERMINATED BY ',' MAP KEYS TERMINATED BY ':'");
+		String localPath = getClass().getResource("/test-data/all_types_table.txt").getFile();
+		clusterAndHiveResource.copyLocalFileToHiveGateWay(localPath,
+				"/tmp/test-data/all_types_table.txt");
+		clusterAndHiveResource.execHiveSql("load data local inpath '/tmp/test-data/all_types_table.txt' " +
+											"into table all_types_table");
+		clusterAndHiveResource.execHiveSql("CREATE TABLE dest_all_types_table like all_types_table");
+		try (final ClusterController clusterController = flinkResource.startCluster(1)) {
+			JobSubmission.JobSubmissionBuilder jobSubmissionBuilder = new JobSubmission.JobSubmissionBuilder(testJarPath);
+			jobSubmissionBuilder.setParallelism(1)
+					.addOption("-ys", "1")
+					.addOption("-ytm", "1000")
+					.addOption("-yjm", "1000")
+					.addOption("-c", HiveReadWriteDataExample.class.getCanonicalName())
+					.addArgument("--hiveVersion", hiveVersion)
+					.addArgument("--sourceTable", "all_types_table")
+					.addArgument("--targetTable", "dest_all_types_table");
```

Comment:
I think the job submission code is the same for the 2 test cases. Can we reuse it?